Sources/Boundaries/Store.swift

```diff
@@ -52,4 +39,85 @@ public final class Store<S, A> {
   public func subscribe<T>(state keyPath: KeyPath<S, T>, _ subscriber: @escaping (T) -> Void) {
     self.subscribe(subscriber <<< view(getting(keyPath)))
   }
+
+  // Runs the effects and dispatches the resulting actions.
+  private func interpret(_ effect: Cmd<E>) {
+    switch effect {
+    case let .execute(e):
+      if let action = self.execute(e) {
+        self.dispatch(action)
+      }
+
+    case .parallel:
+      DispatchQueue.global().async {
+        catOptionals(self.interpretedActions(effect))
+          .forEach { action in
+            DispatchQueue.main.async {
+              self.dispatch(action)
+            }
+        }
+      }
+
+    case let .dispatch(action):
+      self.dispatch(action)
+
+    case let .sequence(effects):
+      effects.forEach(self.interpret)
+    }
+  }
+
+  // Runs the effects and collects all of the resulting actions without dispatching them.
+  private func interpretedActions(_ effect: Cmd<E>) -> [A?] {
+    switch effect {
+    case let .execute(e):
+      return [self.execute(e)]
+
+    case let .parallel(effects):
+      return effects.pmap(self.interpretedActions).flatMap(id)
```

Comment:
add a `pmap` helper that breaks up an array into chunks based on number of cores, transforms them in parallel, and then collects them together.

Reply:
Neat!